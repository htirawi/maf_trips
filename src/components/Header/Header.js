import { useState, useRef } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import logo from "../../assets/images/logo.svg";

const Header = (props) => {
    const [isShowEle, setIsShowEle] = useState(false);
    const { refs } = props;

    const navInput = useRef(null);
    const handleClick = () => {
        navInput.current.click();
        setIsShowEle(true);
    };

    const goToFooter = (id) => {
        const reference = refs[`${id}`];
        navInput.current.click();
        setIsShowEle((preShowElem) => {
            return preShowElem ? !preShowElem : preShowElem;
        });
        setTimeout(() => {
            if (!reference?.current) return;
            [1, 2, 3, 4].forEach((elem) => {
                refs[`${elem}`].current.classList.remove("active");
            });
            reference.current.classList.add("active");
            reference.current.scrollIntoView({ behavior: "smooth" });
            reference.current.click({ behavior: "smooth" });
        }, 400);
    };

    return (
        <>
            <header className='flex justify-between xs:mx-2'>
                <a href='/'>
                    <img src={logo} alt='ntLogo' className='w-16 h-16'></img>
                </a>

                <div className='flex items-center space-x-4'>
                    <input
                        type='text'
                        className='bg-transparent border-none px-8 rounded-full w-11/12 py-3 focus:outline-none focus:ring-2 focus:w-100'
                        placeholder='Search trips'
                    ></input>
                    <div
                        className='flex items-center space-x-4'
                        aria-label='search'
                    >
                        <FontAwesomeIcon
                            icon={faSearch}
                            className='text-white'
                        />
                    </div>
                    <button
                        onClick={() => handleClick()}
                        type='button'
                        className='flex
                            items-center
                            justify-center
                            w-40 h-14
                            cursor-pointer
                            rounded-full
                            bg-custom-cyan
                            text-white
                            text-base
                            font-semibold
                            font-larsseit
                            uppercase
                            lg:w-56
                            focus:outline-none
                            '
                        aria-label='Menu'
                    >
                        menu
                    </button>
                    <div
                        className='navigation'
                        style={{
                            visibility: isShowEle ? "visible" : "hidden",
                        }}
                    >
                        <input
                            type='checkbox'
                            className='navigation__checkbox'
                            id='navi-toggle'
                            ref={navInput}
                        ></input>
                        <label
                            htmlFor='navi-toggle'
                            className='navigation__button'
                            onClick={() => setIsShowEle(false)}
                        >
                            <span className='navigation__icon'>&nbsp;</span>
                        </label>
                        <div className='navigation__background'>&nbsp;</div>
                        <nav className='navigation__nav'>
                            <ul className='navigation__list'>
                                <li
                                    className='navigation__item'
                                    onClick={() => {
                                        goToFooter(1);
                                    }}
                                >
                                    <a href='!#' className='navigation__link'>
                                        <span>01</span>Chill Adventure
                                    </a>
                                </li>
                                <li
                                    className='navigation__item'
                                    onClick={() => {
                                        goToFooter(2);
                                    }}
                                >
                                    <a href='!#' className='navigation__link'>
                                        <span>02</span>Spooky Times
                                    </a>
                                </li>
                                <li
                                    className='navigation__item'
                                    onClick={() => {
                                        goToFooter(3);
                                    }}
                                >
                                    <a href='!#' className='navigation__link'>
                                        <span>03</span>Desert Madness
                                    </a>
                                </li>
                                <li
                                    className='navigation__item'
                                    onClick={() => {
                                        goToFooter(4);
                                    }}
                                >
                                    <a href='!#' className='navigation__link'>
                                        <span>04</span>Out in the wild
                                    </a>
                                </li>
                            </ul>
                        </nav>
                    </div>
                </div>
            </header>
        </>
    );
};

export default Header;
