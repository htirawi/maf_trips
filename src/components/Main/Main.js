import { useState, useRef, useEffect } from "react";
import Header from "../Header/Header";
import Content from "../Content/Content";
import Footer from "../Footer/Footer";
import TripOptions from "../TripOptions/TripOptions";
import BackgroundImage from "../../assets/images/ganapathy.png";

const Main = () => {
    const chillRef = useRef(null);
    const spookyRef = useRef(null);
    const dessertRef = useRef(null);
    const outWildRef = useRef(null);
    const scrollToContent = useRef(null);
    const [container, setContainer] = useState(window.innerHeight);

    /* we are using these values to reflect tabs names */
    const refs = {
        1: chillRef,
        2: spookyRef,
        3: dessertRef,
        4: outWildRef,
        5: scrollToContent,
    };
    /* this function is for dynamically set the height of the page */
    const handleContainerHeight = () => {
        setContainer(window.innerHeight);
    };
    useEffect(() => {
        window.addEventListener("resize", handleContainerHeight);
    }, []);
    return (
        <>
            <div
                className='bg-cover -2x relative'
                style={{
                    backgroundImage: `url(${BackgroundImage})`,
                    height: container,
                }}
            >
                {/* Main container */}
                <div className='max-w-7xl mx-auto sm:px-6 lg:px-4 pt-10'>
                    <Header refs={refs} />
                    <Content />
                    <Footer refs={scrollToContent} />
                    {/* this div is the one that responsible for the shadow at the end of main poge image */}
                    <div className='bg-custom-gradient-to-r absolute w-full bottom-0 z-0 left-0 h-96'></div>
                </div>
                <TripOptions refs={refs} />
            </div>
        </>
    );
};

export default Main;
