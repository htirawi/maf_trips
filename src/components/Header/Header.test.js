import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { shallow } from "enzyme";
import Header from "../../../src/components/Header/Header";
import Main from "../../../src/components/Main/Main";
import toJson from "enzyme-to-json";

afterEach(cleanup);

describe("<Header/> component should render successfully", () => {
    it("should render the component", () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toHaveLength(1);
    });
});

it("<Header/> renders without crashing given the required props", () => {
    const props = {
        isShowEle: false,
    };
    const wrapper = shallow(<Header {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});
