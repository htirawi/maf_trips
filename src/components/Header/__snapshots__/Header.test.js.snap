// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing given the required props 1`] = `
<Fragment>
  <header
    className="flex justify-between xs:mx-2"
  >
    <a
      href="/"
    >
      <img
        alt="ntLogo"
        className="w-16 h-16"
        src="logo.svg"
      />
    </a>
    <div
      className="flex items-center space-x-4"
    >
      <input
        className="bg-transparent border-none px-8 rounded-full w-11/12 py-3 focus:outline-none focus:ring-2 focus:w-100"
        placeholder="Search trips"
        type="text"
      />
      <div
        aria-label="search"
        className="flex items-center space-x-4"
      >
        <FontAwesomeIcon
          border={false}
          className="text-white"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f002",
                "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
              ],
              "iconName": "search",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
      <button
        aria-label="Menu"
        className="flex
 items-center
 justify-center
 w-40 h-14
 cursor-pointer
 rounded-full
 bg-custom-cyan
 text-white
 text-base
 font-semibold
 font-larsseit
 uppercase
 lg:w-56
 focus:outline-none
 "
        onClick={[Function]}
        type="button"
      >
        menu
      </button>
      <div
        className="navigation"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <input
          className="navigation__checkbox"
          id="navi-toggle"
          type="checkbox"
        />
        <label
          className="navigation__button"
          htmlFor="navi-toggle"
          onClick={[Function]}
        >
          <span
            className="navigation__icon"
          >
             
          </span>
        </label>
        <div
          className="navigation__background"
        >
           
        </div>
        <nav
          className="navigation__nav"
        >
          <ul
            className="navigation__list"
          >
            <li
              className="navigation__item"
              onClick={[Function]}
            >
              <a
                className="navigation__link"
                href="!#"
              >
                <span>
                  01
                </span>
                Chill Adventure
              </a>
            </li>
            <li
              className="navigation__item"
              onClick={[Function]}
            >
              <a
                className="navigation__link"
                href="!#"
              >
                <span>
                  02
                </span>
                Spooky Times
              </a>
            </li>
            <li
              className="navigation__item"
              onClick={[Function]}
            >
              <a
                className="navigation__link"
                href="!#"
              >
                <span>
                  03
                </span>
                Desert Madness
              </a>
            </li>
            <li
              className="navigation__item"
              onClick={[Function]}
            >
              <a
                className="navigation__link"
                href="!#"
              >
                <span>
                  04
                </span>
                Out in the wild
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</Fragment>
`;
